packages/venia-ui/lib/RootComponents/CMS/cms.js

```diff
@@ -49,35 +49,44 @@ const CMSPage = props => {
         setPageLoading(false);
     }
 
-    const { content_heading, title } = data.cmsPage;
+    const {
+        content_heading,
+        title,
+        meta_title,
+        meta_description,
+        content
+    } = data.cmsPage;
 
     const headingElement =
         content_heading !== '' ? (
             <h1 className={classes.heading}>{content_heading}</h1>
         ) : null;
 
-    let content;
+    const pageTitle = meta_title && meta_title.length > 0 ? meta_title : title;
+
+    let pageContent;
     // Only render <RichContent /> if the page isn't empty and doesn't contain the default CMS Page text.
     if (
-        data.cmsPage.content &&
-        data.cmsPage.content.length > 0 &&
-        !data.cmsPage.content.includes('CMS homepage content goes here.')
+        content &&
+        content.length > 0 &&
+        !content.includes('CMS homepage content goes here.')
```

Comment:
This is probably a good opportunity to create a talon for this component, given that we already have them for the other root components. I don't think that really expands the scope here, just moves the code around. And it strikes me as the right thing to do. Thoughts?

Normally these kinds of derived booleans would be returned by a talon as `hasContent` and `isHomepage`, or some such.

Reply:
This code was always intended as a "hack" to get the Page Builder code live, I think rather than throwing the hack into a talon now we should have a future discussion on how to handle this correctly. As checking the contents of a string to conduct logic is generally a bad time and could easily produce incorrect results.

In my opinion PWA Studio should render "CMS homepage content goes here." rather than replacing this with the `<CategoryList />`, potentially this is an opportunity for us to build out the sample data for the default page (without Page Builder), to be able to remove this logic.

Reply:
As agreed I moved this logic into a talon.